packages/wallets/xdefi/src/walletHelpers.ts

```diff
@@ -77,7 +77,12 @@ const transaction = async ({
   params: TransactionParams[] | any;
   chain: Chain;
 }): Promise<string> => {
-  const client = method === "deposit" ? window.xfi?.thorchain : getXDEFIProvider(chain);
+  const client =
+    method === "deposit"
+      ? chain === Chain.Maya
+        ? window.xfi?.mayachain
+        : window.xfi?.thorchain
+      : getXDEFIProvider(chain);
```

Comment:
do we still need this hardcoded case here for deposit?

Reply:
@phips28 for now yes - but we'll withdraw provider specific logic to plugins and introduce last breaking before stable v1 but it will improve going further with chainflip, maya and more :)